### ISSUE TITLE
response content type is ignored

### ISSUE THREAD

**Issue by casshill** [DiUS/pact-jvm#312]
I'm using pact-jvm-consumer-groovy (au.com.dius:pact-jvm-consumer-groovy_2.11:2.4.15) and in the examples in the readme it shows that you can specify the expected content type like this:
      
willRespondWith(
          status: 200,
          headers: ['Content-Type': 'application/hal+json']
      )

I can't get this to work. I want the content type to be application/hal+json, but every time my pact file is produced it changes it to application/json, which is not what I specified.
It looks like this:

            "response": {
                "status": 200,
                "headers": {
                    "Content-Type": "application/json"
                },
...
"pact-specification": {
            "version": "2.0.0"
        },
        "pact-jvm": {
            "version": "2.4.15"
        }

Is there a way to specify the content type as hal+json instead of json?

**Comment by uglyog** [comment]
You should be able to set the content type to anything you need. Let me have a look to see whyt it is being overridden.

**Comment by uglyog** [comment]
Sorry, I can't replicate this. When I set the content type, it appears in the pact file with what I set it.

Could you provide the entire test you are running?

**Comment by casshill** [comment]
Here's my test:

    def 'requests products with facets'() {
      given:
      def subscriber = new TestSubscriber<CategoriesEntity>()
      def pactBuilder = new PactBuilder()
      def service = buildService(ProductsService.class)
      def request_gender = 'male'

    pactBuilder {
      serviceConsumer 'consumerName'
      hasPactWith 'providerName'
      port PORT

      uponReceiving 'A request for products'
      withAttributes(
          method: 'GET',
          path: "/catalog/products",
          headers: ['Accept': 'application/json'],
          query: [
              gender: request_gender
          ])

      willRespondWith(
          status: 200,
          headers: ['Content-Type': 'application/hal+json']
      )
      withBody {...}
    }

and here's my pact file:

```
  {
    "provider": {
        "name": "providerName"
    },
    "consumer": {
        "name": "consumerName"
    },
    "interactions": [
        {
            "description": "A request for products",
            "request": {
                "method": "GET",
                "path": "/catalog/products",
                "headers": {
                    "Accept": "application/json"
                },
                "query": "gender=male"
            },
            "response": {
                "status": 200,
                "headers": {
                    "Content-Type": "application/json"
                },
                "body": {...}
    ...
   "metadata": {
        "pact-specification": {
            "version": "2.0.0"
        },
        "pact-jvm": {
            "version": "2.4.15"
        }
    }
```

You can see the pact file has application/json as the content type even though I specified hal+json. If I remove the content type from the response in my test ie.

```
      willRespondWith(
          status: 200,
          headers: ['Content-Type': 'application/hal+json']
      )
```

is now

```
      willRespondWith(
          status: 200
      )
```

My pact file is still the same, it still puts the application/json as the content type in the pact file. 
I'm guessing there is some sort of default set and it is overriding the value I set.

**Comment by uglyog** [comment]
Ok, the `withBody` is resetting the content type. I'll fix that, but in the short term you can tell the `withBody` method the content type to use with:

```groovy
withBody(mimeType: 'application/hal+json') {
  // ...
}
```

**Comment by casshill** [comment]
Great thanks I'll use the above until the fix has been made.

**Comment by uglyog** [comment]
Fix released with version 3.3.1
